app/src/main/java/com/p2p/data/UserInfoStorage.kt

```diff
@@ -0,0 +1,11 @@
+package com.p2p.data
+
+/** This interface brings the capacity to save user info into some storage. */
+interface UserInfoStorage {
```

Comment:
¿Para qué esto es una interfaz? No entiendo qué otra implementación puede tener.

Reply:
Meh, otras implementaciones puede haber como una sesión que dure solo mientras viva la app (sería un object). Además te sorprendería lo mucho que cambian las cosas en Android jajaj, capaz a mitad de año les pinte recomendar no usar SharedPreferences y empezar a recomendar PepitoPreferences (?).

Más allá de eso, esto es por las siguientes cosas teóricas:
- [Coding to interfaces, not implementation](https://medium.com/javarevisited/oop-good-practices-coding-to-the-interface-baea84fd60d3): básicamente, que los clientes conozcan interfaces y no directamente las implementaciones.
- [Clean architecture](https://blog.cleancoder.com/uncle-bob/2012/08/13/the-clean-architecture.html): dice que la arquitectura debería ser como una cebollita en donde las capas de adentro no conozcan las cosas de afuera. Yo no implementé algunas capas para no sumar tantas cosas, pero la parte de que los view models no conozcan nada de Android me parece algo bueno de respetar para mantener las responsabilidades separadas.
![image](https://user-images.githubusercontent.com/65614580/117671130-dad01e00-b17e-11eb-895a-9717436c64d8.png)